devicesetup/src/main/java/io/particle/android/sdk/devicesetup/ui/DiscoverDeviceActivity.java

```diff
@@ -296,17 +339,15 @@ protected SetupStepException doInBackground(Void... voids) {
             @Override
             protected void onPostExecute(SetupStepException error) {
                 connectToApTask = null;
-                if (error == null) {
+                if (error == null || (BaseActivity.setupOnly && error instanceof DeviceAlreadyClaimed)) {
```

Comment:
This is one of the changes from that external pull request that allows for just claiming